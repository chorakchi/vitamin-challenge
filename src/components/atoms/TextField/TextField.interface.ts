import { InputHTMLAttributes } from 'react';


export interface TextFieldProps
  extends InputHTMLAttributes<HTMLInputElement> {
    label?: string;
    error?: boolean;
}