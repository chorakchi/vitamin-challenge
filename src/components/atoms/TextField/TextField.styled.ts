import styled from "styled-components";

export const TextFieldStyled = styled.input<{ error: boolean }>`
  background-color: white;
  border-radius: 8px;
  margin: 16px 0;
  padding: 20px;
  font-size: 18px;
  position: relative;
  width: 208px;
  height: 40px;
  /* border-color: ${({error}) => error ? 'red' : 'var(--primary-color)'}; */
`;
