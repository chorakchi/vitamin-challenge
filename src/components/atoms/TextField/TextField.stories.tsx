import { ComponentStory, ComponentMeta } from "@storybook/react";

import { TextField } from "./TextField";

export default {
  title: "Atoms/TextField",
  component: TextField,
  parameters: {
    docs: {
      description: {
        component:
          `Text fields let users enter and edit text.<br/>
                Text fields allow users to enter text into a UI. They typically appear in forms and dialogs.
                <br/>` +
          `The TextField  component is a complete form control including a label, input. It comes with two variants: outlined (default), and primary.<br/>` +
          `<br/> <br/> <h5> Form props:</h5>` +
          " Standard form attributes are supported e.g. ```required```, ```disabled```, ```type```, etc. as well as a ```helperText``` which is used to give context about a field's input, such as how the input will be used. <br/> <br/>"
      },
    },
    backgrounds: {
      default: "light",
    },
  },
} as ComponentMeta<typeof TextField>;

const Template: ComponentStory<typeof TextField> = (args) => (
  <TextField {...args} />
);

export const TextField_ = Template.bind({});
TextField_.args = {};
