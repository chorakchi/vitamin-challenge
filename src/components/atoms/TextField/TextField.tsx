import { TextFieldStyled } from './TextField.styled';
import { TextFieldProps } from './TextField.interface';
import { Typography } from '../Typography/Typography';
import { Stack } from 'components';
export const TextField: React.FC<TextFieldProps> = ({
  label="Label:",
  error,
    ...props
}: TextFieldProps) => {
    return (
        <Stack direction="column">
            <Typography as="label" htmlFor={props.id}>
                {label}
                <br/>
                <TextFieldStyled
                error={error}
                {...props}
                />
            </Typography>
        </Stack>
    );
};
