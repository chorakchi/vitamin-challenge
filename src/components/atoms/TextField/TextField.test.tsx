import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import {TextField} from "./TextField";

describe("TextField", () => {
  it("should render the TextField", () => {
    const subject = shallow(<TextField />);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<TextField />).toJSON();
    expect(subject).toMatchSnapshot();
  });
});