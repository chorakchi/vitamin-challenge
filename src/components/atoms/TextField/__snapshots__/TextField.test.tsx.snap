// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should render correctly and match to the Snapshot 1`] = `
<div
  className="Stackstyled__StackStyled-sc-1adxq1u-0 bnrRyH"
  direction="column"
>
  <label
    className="Typographystyled__TypographyStyled-sc-i22hyo-0 iKdygM"
  >
    Label:
    <br />
    <input
      className="TextFieldstyled__TextFieldStyled-sc-mvtmc2-0 NTHeK"
    />
  </label>
</div>
`;
